common/flatpak-installation.c

```diff
@@ -1092,7 +1097,67 @@ flatpak_installation_list_installed_refs_for_update (FlatpakInstallation *self,
       /* Note: local_commit may be NULL here */
       if (remote_commit != NULL &&
           g_strcmp0 (remote_commit, local_commit) != 0)
-        g_ptr_array_add (updates, g_object_ref (installed_ref));
+        {
+          g_ptr_array_add (updates, g_object_ref (installed_ref));
+        }
+      else if (g_str_has_prefix (full_ref, "app/"))
+        {
+          g_autoptr (GPtrArray) installed_related_refs = NULL;
```

Comment:
This big block of code could do with one or two comments explaining what it does, at a high level.

Reply:
I already wrote the high-level gist of what it does as the part of the API documentation. Is it good to have more fine-grained explanation here as well?

Reply:
IMO, yes, because it’s quite a long function. But the actual flatpak maintainers might have different ideas.